src/libcore/num/mod.rs

```diff
@@ -50,6 +50,7 @@ assert_eq!(size_of::<Option<core::num::", stringify!($Ty), ">>(), size_of::<", s
                 #[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
                 #[repr(transparent)]
                 #[rustc_layout_scalar_valid_range_start(1)]
+                #[cfg_attr(not(stage0), rustc_nonnull_optimization_guaranteed)]
```

Comment:
Could this just reuse the attribute in the line above?

Reply:
I think the goal was to separate "performing the optimization" from "guaranteeing the optimization".